packages/web-app-files/tests/unit/helpers/collaborators.spec.js

```diff
@@ -0,0 +1,261 @@
+      input => {
+        // assume a case when share permission is not allowed, the resource is not folder and provided bitmask is 17
+        // two roles with following permissions are available from roleDefinition helper
+        // - viewer with permissions ['read']
+        // - editor with permissions ['read', 'update']
+        // with any of the role, the sum of permission bitmask (1 || 3) does not matches provided bitmask (17)
+        // so the function will return {advancedRole}
+        expect(bitmaskToRole(input.bitmask, input.isFolder, input.allowSharePerm)).toMatchObject({
+          name: 'advancedRole'
+        })
+      }
+    )
+    it.each([
+      {
+        bitmask: 17,
+        isFolder: false,
+        allowSharePerm: true,
+        expectedRole: 'viewer',
+        expectedPermissions: ['read', 'share']
+      },
+      {
+        bitmask: 17,
+        isFolder: false,
+        allowSharePerm: true,
+        expectedRole: 'viewer',
+        expectedPermissions: ['read', 'share']
+      },
+      {
+        bitmask: 3,
+        isFolder: false,
+        allowSharePerm: false,
+        expectedRole: 'editor',
+        expectedPermissions: ['read', 'update']
+      },
+      {
+        bitmask: 15,
+        isFolder: true,
+        allowSharePerm: false,
+        expectedRole: 'editor',
+        expectedPermissions: ['read', 'update', 'create', 'delete']
+      },
+      {
+        bitmask: 31,
+        isFolder: true,
+        allowSharePerm: true,
+        expectedRole: 'editor',
+        expectedPermissions: ['read', 'update', 'create', 'delete', 'share']
+      }
+    ])(
+      'should return the first matching role if the given bitmask equals the role permissions bitmask',
+      input => {
+        // assume a case when share permission is allowed and resource is not folder and provided bitmask is 17
+        // two roles with following permissions are available from the roleDefinition helper
+        // - viewer with permissions ['read', 'share']
+        // - editor with permissions ['read', 'update', 'share']
+        // with the first role, the sum of permission bitmask (1 + 16) matches provided bitmask (17)
+        // so the function has found a match and returns the role {viewer}
+        expect(bitmaskToRole(input.bitmask, input.isFolder, input.allowSharePerm)).toMatchObject({
+          name: input.expectedRole,
+          permissions: input.expectedPermissions
+        })
+      }
+    )
+    it.each(['17', 17.2, '17.8', ['17.8']])('should parse provided bitmask to integer', bitmask => {
+      expect(bitmaskToRole(bitmask, false, true)).toMatchObject({
+        name: 'viewer'
+      })
+    })
+    it.each([true, 0, {}])('should return advanced role for invalid bitmask values', bitmask => {
+      expect(bitmaskToRole(bitmask, false, true)).toMatchObject({
+        name: 'advancedRole'
+      })
+    })
```

Comment:
@kulmann is that what we want? or should there be an error thrown for invalid bitmasks?

Reply:
advanced role for the `0` bitmask would be fine. That bitmask doesn't make too much sense, but it's still valid. For the wrong types there should be some error.

Reply:
thrown error if wrong types are supplied. please review